### PR TITLE
[FLINK-16966][metrics][infuxdb] Convert InfluxDB reporter to plugin

### PR DESCRIPTION
**Pull request by complone**
## What is the purpose of the change

*Convert InfluxDB reporter to plugin,Set SPI configuration for reporter*

## Brief change log

*(for example:)*
  - *The TaskInfo is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *TaskManagers retrieve the TaskInfo from the blob cache*

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Manually verified the change by running a 4 node cluser with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-metrics/flink-metrics-influxdb/src/main/java/org/apache/flink/metrics/influxdb/InfluxdbReporterFactory.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package org.apache.flink.metrics.influxdb;
```
Comment (zentol) [inline flink-metrics/flink-metrics-influxdb/src/main/java/org/apache/flink/metrics/influxdb/InfluxdbReporterFactory.java:1]: This file is missing the apache license header.

File: flink-metrics/flink-metrics-influxdb/src/main/java/org/apache/flink/metrics/influxdb/InfluxdbReporterFactory.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import org.apache.flink.metrics.reporter.MetricReporter;
+import org.apache.flink.metrics.reporter.MetricReporterFactory;
+
+import java.util.Properties;
+
+public class InfluxdbReporterFactory implements MetricReporterFactory {
```
[the remark this answers is not in the document]
Comment (complone) [reply flink-metrics/flink-metrics-influxdb/src/main/java/org/apache/flink/metrics/influxdb/InfluxdbReporterFactory.java:8]: OK, I'll do it today

File: flink-metrics/flink-metrics-influxdb/src/main/java/org/apache/flink/metrics/influxdb/InfluxdbReporterFactory.java (hunk 1 above)
Comment (complone) [reply flink-metrics/flink-metrics-influxdb/src/main/java/org/apache/flink/metrics/influxdb/InfluxdbReporterFactory.java:1]: OK, I'll do it today

Comment (complone) [reply flink-metrics/flink-metrics-influxdb/src/main/java/org/apache/flink/metrics/influxdb/InfluxdbReporterFactory.java:19]: @zentol Apache license added

File: flink-metrics/flink-metrics-influxdb/src/main/java/org/apache/flink/metrics/influxdb/InfluxdbReporterFactory.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (complone) [reply flink-metrics/flink-metrics-influxdb/src/main/java/org/apache/flink/metrics/influxdb/InfluxdbReporterFactory.java:26]: @Myasuka Apache license added

[the remark this answers is not in the document]
Comment (Myasuka) [reply flink-metrics/flink-metrics-influxdb/src/main/java/org/apache/flink/metrics/influxdb/InfluxdbReporterFactory.java:26]: I'm afraid you did not understand what I mean. This class `InfluxdbReporterFactory` missed a javadoc, please check the CI failure or build your code local.
Moreover, I still suggest you to fill the content of description of this PR.

[the remark this answers is not in the document]
Comment (complone) [reply flink-metrics/flink-metrics-influxdb/src/main/java/org/apache/flink/metrics/influxdb/InfluxdbReporterFactory.java:29]: > I'm afraid you did not understand what I mean. This class `InfluxdbReporterFactory` missed a javadoc, please check the CI failure or build your code local.
> Moreover, I still suggest you to fill the content of description of this PR.


Added description for pr

**Comment by complone** [comment]
> ## 变更的目的是什么
> _（例如：此请求请求使任务部署通过blob服务器进行，​​而不是通过RPC。这样一来，我们避免在每次部署时都将它们重新转移（在恢复过程中。）_
> 
> ## 简要变更日志
> _（例如：）_
> 
> * _TaskInfo在创建作业时作为持久工件存储在Blob存储中_
> * _部署RPC仅传输Blob存储参考_
> * _TaskManager从Blob缓存中检索TaskInfo_
> 
> ## 验证此更改
> _（请选择以下任一选项）_
> 
> 此更改是简单的返工/代码清除，没有任何测试内容。
> 
> _（要么）_
> 
> 现有测试已经覆盖了此更改，例如_（请描述测试）_。
> 
> _（要么）_
> 
> 此更改增加了测试，可以进行如下验证：
> 
> _（例：）_
> 
> * _添加了集成测试，以用于具有大负载（100MB）的端到端部署_
> * _扩展集成测试，用于在主服务器（JobManager）故障后进行恢复_
> * _添加的测试可验证TaskInfo在恢复中仅传输一次_
> * _通过运行带有2个JobManager和4个TaskManager，一个有状态的流程序的4节点cluser，并在执行过程中杀死一个JobManager和两个TaskManager来手动验证更改，以验证恢复是否正确进行。_
> 
> ## 此拉取请求是否可能影响以下部分之一：
> * 依赖关系（是否添加或升级了依赖关系）：（是/否）
> * 公共API，即，任何已更改的类都带有`@Public(Evolving)`：（是/否）
> * 序列化器：（是/否/不知道）
> * 运行时按记录的代码路径（对性能敏感）：（是/否/不知道）
> * 任何影响部署或恢复的因素：JobManager（及其组件），检查点，Kubernetes / Yarn / Mesos，ZooKeeper ：（是/否/不知道）
> * S3文件系统连接器：（是/否/不知道）
> 
> ## 文献资料
> * 此拉取请求是否引入了新功能？（是/否）
> * 如果是，该功能如何记录？（不适用/ docs / JavaDocs /未记录）

Added description for pr

**Pull request merged**